Sync `DeviceMotionEvent` as per web specification

Ahmad-S792 opened the pull request:
#### b75a96c8dc729226d94e528f614e6b7c96a914c7
<pre>
Sync `DeviceMotionEvent` as per web specification
<a href="https://bugs.webkit.org/show_bug.cgi?id=271098">https://bugs.webkit.org/show_bug.cgi?id=271098</a>
<a href="https://rdar.apple.com/problem/125271931">rdar://problem/125271931</a>

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Blink / Chrome and Web Specification [1]:

[1] <a href="https://w3c.github.io/deviceorientation/#device-motion-event-api">https://w3c.github.io/deviceorientation/#device-motion-event-api</a>

By removing `initDeviceMotionEvent`, WebKit aligns with Web Specification
and Blink also removed it in 2017 via below commit:

Reference: <a href="https://chromium.googlesource.com/chromium/src.git/+/0921e8319f228e406019b434701ac9f181ee8ea1">https://chromium.googlesource.com/chromium/src.git/+/0921e8319f228e406019b434701ac9f181ee8ea1</a>

Additionally, it also make &apos;interval&apos; non-nullable. Also adding &apos;=null&apos;
values for &apos;Accelerate&apos; and &apos;Rotate&apos; dictonaries.

Additionally, we now also add `constructor`, which had FIXME.

* Source/WebCore/dom/DeviceMotionData.cpp:
(WebCore::DeviceMotionData::create):
(WebCore::DeviceMotionData::DeviceMotionData):
* Source/WebCore/dom/DeviceMotionData.h:
(WebCore::DeviceMotionData::interval const):
* Source/WebCore/dom/DeviceMotionEvent.cpp:
(WebCore::convert):
(WebCore::DeviceMotionEvent::interval const):
(WebCore::DeviceMotionEvent::create):
(WebCore::DeviceMotionEvent::DeviceMotionEvent):
(WebCore::DeviceMotionEvent::initDeviceMotionEvent): Deleted.
* Source/WebCore/dom/DeviceMotionEvent.h:
* Source/WebCore/dom/DeviceMotionEvent.idl:
* LayoutTests/fast/dom/DeviceMotion/optional-event-properties.html: Rebaselined
* LayoutTests/fast/dom/DeviceMotion/optional-event-properties-expected.txt: Rebaselined
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b75a96c8dc729226d94e528f614e6b7c96a914c7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/148809 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/21522 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/157494 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/102239 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/e80220cb-88a6-472d-8257-4dc8e94a6522) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/150682 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/21974 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/21400 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/157494 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/102239 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/1467338e-cbb7-49b8-8481-978637d023c6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/151769 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/155/builds/21974 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/157494 "Hash b75a96c8 for PR 26001 does not build (failure)") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/c726ac8c-cef8-4fa4-b74d-71152f3bf21b) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/155/builds/21974 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/5259 "Hash b75a96c8 for PR 26001 does not build (failure)") | | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/155/builds/21974 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/159832 "Hash b75a96c8 for PR 26001 does not build (failure)") | | 
| | [❌ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/2969 "Failed to build and analyze WebKit") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/159832 "Hash b75a96c8 for PR 26001 does not build (failure)") | | 
| | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/21324 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/156/builds/21400 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/159832 "Hash b75a96c8 for PR 26001 does not build (failure)") | | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/21332 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/77520 "Hash b75a96c8 for PR 26001 does not build (failure)") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/168/builds/15091 "Hash b75a96c8 for PR 26001 does not build (failure)") | | | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/20934 "Hash b75a96c8 for PR 26001 does not build (failure)") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/84736 "Failed to build and analyze WebKit") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/20666 "Hash b75a96c8 for PR 26001 does not build (failure)") | | | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/20813 "Hash b75a96c8 for PR 26001 does not build (failure)") | | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/20722 "Hash b75a96c8 for PR 26001 does not build (failure)") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/DeviceMotionEvent.idl:
```diff
@@ -23,24 +23,21 @@
     Exposed=Window
 ] interface DeviceMotionEvent : Event {
+    // FIXME: Add `constructor` by uncommenting below.
+    // constructor(DOMString type, optional DeviceMotionEventInit eventInitDict = {});
```
We should just add constructor.

karlcow replied:
@Ahmad-S792 did you have a chance to check this? Or do you need help. 
I'm asking because of the recent bug
https://bugs.webkit.org/show_bug.cgi?id=271891

Ahmad-S792 replied:
I haven't looked into it yet but I manage to fix another `[SecureContext]` issue in these though. So some progress but not directly on `constructor` side. :-)

Ahmad-S792 replied:
@karlcow - I just rebased it and added few more alignments but still working on how to add 'constructor'. Although it did progressed one or two WPT tests.

Ahmad-S792 committed:
Sync `DeviceMotionEvent` as per web specification
https://bugs.webkit.org/show_bug.cgi?id=271098
rdar://problem/125271931

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Blink / Chrome and Web Specification [1]:

[1] https://w3c.github.io/deviceorientation/#device-motion-event-api

By removing `initDeviceMotionEvent`, WebKit aligns with Web Specification
and Blink also removed it in 2017 via below commit:

Reference: https://chromium.googlesource.com/chromium/src.git/+/0921e8319f228e406019b434701ac9f181ee8ea1

Additionally, it also make 'interval' non-nullable. Also adding '=null'
values for 'Accelerate' and 'Rotate' dictonaries.

Additionally, we now also add `constructor`, which had FIXME.

* Source/WebCore/dom/DeviceMotionData.cpp:
(WebCore::DeviceMotionData::create):
(WebCore::DeviceMotionData::DeviceMotionData):
* Source/WebCore/dom/DeviceMotionData.h:
(WebCore::DeviceMotionData::interval const):
* Source/WebCore/dom/DeviceMotionEvent.cpp:
(WebCore::convert):
(WebCore::DeviceMotionEvent::interval const):
(WebCore::DeviceMotionEvent::create):
(WebCore::DeviceMotionEvent::DeviceMotionEvent):
(WebCore::DeviceMotionEvent::initDeviceMotionEvent): Deleted.
* Source/WebCore/dom/DeviceMotionEvent.h:
* Source/WebCore/dom/DeviceMotionEvent.idl:
* LayoutTests/fast/dom/DeviceMotion/optional-event-properties.html: Rebaselined
* LayoutTests/fast/dom/DeviceMotion/optional-event-properties-expected.txt: Rebaselined

rniwa commented on Source/WebCore/dom/DeviceMotionEvent.idl:
```diff
@@ -29,23 +29,14 @@
 
     [CallWith=CurrentDocument, EnabledBySetting=DeviceOrientationPermissionAPIEnabled] static Promise<DeviceOrientationOrMotionPermissionState> requestPermission();
-
-    undefined initDeviceMotionEvent(optional [AtomString] DOMString type = "",
```
I don't think we can get rid of this function. There could be websites that rely on it.

rniwa reviewed (changes requested):
I don't think we can or should remove `initDeviceMotionEvent`.
